srcpkgs/python3-betamax-serializers/template

```diff
@@ -0,0 +1,16 @@
+# Template file for 'python3-betamax-serializers'
+pkgname=python3-betamax-serializers
+version=0.2.1
+revision=1
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-betamax"
+short_desc="Serializers for betamax that may eventually be included in betamax"
+maintainer="Jan Christian Grünhage <jan.christian@gruenhage.xyz>"
+license="Apache-2.0"
+homepage="https://gitlab.com/betamax/serializers"
+changelog="https://gitlab.com/betamax/serializers/-/raw/master/HISTORY.rst"
+distfiles="${PYPI_SITE}/b/${pkgname/python3-/}/${pkgname/python3-/}-${version}.tar.gz"
+checksum=345c419b1b73171f2951c62ac3c701775ac4b76e13e86464ebf0ff2a978e4949
+# Has no tests
+make_check=no
```

Comment:
if there are no tests, they usually don't fail, so we don't have to disable them

Reply:
It does try though, and then fails:

```
=> python3-betamax-serializers-0.2.1_1: running do_check ...
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
/usr/lib/python3.11/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
  warnings.warn(
WARNING: The wheel package is not available.
/usr/bin/python3: No module named pip
error: Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmpxidysgrh', '--quiet', 'betamax>=0.3.2']' returned non-zero exit status 1.
=> ERROR: python3-betamax-serializers-0.2.1_1: do_check: '${make_check_pre} python3 setup.py ${make_check_target} ${make_check_args}' exited with 1
=> ERROR:   in do_check() at common/build-style/python3-module.sh:28
```

Do you have a suggestion for a better comment?

Reply:
> `...--quiet', 'betamax>=0.3.2']' returned ...`
It's just checking for required dependencies, in this case betamax, `checkdepends="${depends} python3-yaml"` will help

Reply:
> if there are no tests, they usually don't fail, so we don't have to disable them

Incorrect, the python3-module build style does this all the time, because it attempts to run `python3 setup.py test` regardless of whether tests exist. This command will:
- depend on all runtime dependencies, or attempt to install local copies of them with pip, but NOT all "extras" (optional) dependencies
- then import all source files (including the ones that have optional dependencies, which are not otherwise mentioned in the template since xbps does not support optional dependencies)
- then detect that there are zero tests, and not run them

And setuptools upstream has deprecated `setup.py test`, so any packages that actually used it (which was relatively uncommon, as people tend to prefer pytest or nose or whatever) have been busily porting away from it.

Incidentally, this: https://github.com/void-linux/void-packages/blob/0657a8bea7e6d89cff327e31949b5eebcb883952/common/build-style/python3-module.sh#L21-L23

is logically broken, because it is fallaciously incorrect to print a warning that "setup.py doesn't define a test command". That's not the role of setup.py to do -- the test command is defined globally in setuptools itself regardless of what a setup.py file does (a setup.py file can override the builtin one, the same way it can override the builtin "build_py" or "install" commands).